deployment/common/changeset/example/add_mint_burners_link.go

```diff
@@ -0,0 +1,70 @@
+package example
+
+import (
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/common/changeset"
+)
+
+type AddMintersBurnersLinkConfig struct {
+	ChainSelector uint64
+	Minters       []common.Address
+	Burners       []common.Address
+}
+
+var _ deployment.ChangeSet[*AddMintersBurnersLinkConfig] = AddMintersBurnersLink
+
+// AddMintersBurnersLink grants the minter / burner role to the provided addresses.
+func AddMintersBurnersLink(e deployment.Environment, cfg *AddMintersBurnersLinkConfig) (deployment.ChangesetOutput, error) {
+
+	chain := e.Chains[cfg.ChainSelector]
+	addresses, err := e.ExistingAddresses.AddressesForChain(cfg.ChainSelector)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+	linkState, err := changeset.MaybeLoadLinkTokenChainState(chain, addresses)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+
+	for _, minter := range cfg.Minters {
+		// check if minter is already a minter
+		isMinter, err := linkState.LinkToken.IsMinter(&bind.CallOpts{Context: e.GetContext()}, minter)
```

Comment:
Curious question, how come we are not using the usual `chain.DeployerKey` ?

Reply:
Since this is a read operation, you don't need a private key since its just querying the chain state

Reply:
thanks, make sense, i guess this is mainly for being more secure